fix: update passport-facebook to 3.0

bobbyg603 opened the pull request:
## Template

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

## Summary

passport-facebook's `StrategyOption` constructor param ultimately gets passed to `OAuth2Strategy`.

https://github.com/jaredhanson/passport-facebook/blob/586d535e7db6b972d4c36618762cd0e8125faddb/lib/strategy.js#L56

Depending on the value of `options.state` the `OAuth2Strategy` constructor will set `_stateStore` to either an instance of `PKCEStateStore` or `NonceStore`.

https://github.com/jaredhanson/passport-oauth2/blob/ea9e99adda82dff67502654347589866fea80eb2/lib/strategy.js#L110

This change also updates package to use exports instead of namespaces.

sheetalkamat commented on types/passport-facebook/index.d.ts:
```diff
@@ -1,66 +1,102 @@
-    callbackURL: string;
+declare namespace Strategy {
+    // NOTE: not true for `export import` statements
+    // tslint:disable-next-line:strict-export-declare-modifiers
```
Fix this lint:
Instead of moving these interfaces and classes into namespace, pls pull them out with direct export like before.

bobbyg603 replied:
hi @sheetalkamat, I attempted to fix this issue in [fb28d1e](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/65543/commits/fb28d1e62101fd68028a9b8e25d0d84c2d6b39f4). Is there something else I need to do to appease @DangerBotOSS?

DangerBotOSS commented:
<!--
  0 failure: 
  0 warning: 
  
  1 markdown notices
  DangerID: danger-id-Danger;
-->



Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## passport-facebook ([<kbd>unpkg</kbd>](https://unpkg.com/browse/passport-facebook@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'passport-facebook'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against fb28d1e62101fd68028a9b8e25d0d84c2d6b39f4
</p>

bobbyg603 commented:
Hey @staticfunction, @lucasmotta, I think this is good to go. If either of you have a moment it would be great to get this merged 🙂. Thanks!

andrewbranch commented:
Did the order of basically every declaration change? It’s a little hard to review with the diff showing so many changes.

bobbyg603 commented:
> Did the order of basically every declaration change? It’s a little hard to review with the diff showing so many changes.

Not intentionally, I'll see if I can improve the diff next week.

bobbyg603 commented on types/passport-facebook/index.d.ts:
```diff
@@ -27,40 +31,62 @@ export interface AuthenticateOptions extends passport.AuthenticateOptions {
 }
 
-export interface StrategyOption {
+export interface StrategyOptions {
```
Made plural to follow conventions in passport-oauth2 https://github.com/DefinitelyTyped/DefinitelyTyped/blob/48ea2d936c47ae6cb56b7b0719d0fa2cd30a9612/types/passport-oauth2/index.d.ts#L100

bobbyg603 commented on types/passport-facebook/index.d.ts:
```diff
@@ -27,40 +31,62 @@ export interface AuthenticateOptions extends passport.AuthenticateOptions {
 
-export interface StrategyOptionWithRequest extends StrategyOption {
+
+export interface StrategyOptionsWithRequest extends Omit<StrategyOptions, 'passReqToCallback'> {
```
Using `Omit` instead of `_StrategyOptionsBase`.

bobbyg603 commented:
Hi @andrewbranch! I've updated the PR to improve the diff. I've also rebased on 48ea2d9. There's still a lot of space changes, but I believe that's because the linting rules have changed since the prior version of this package. Please let me know if there's anything else I can do - thanks!

andrewbranch reviewed (approved):
Thanks for the updates @bobbyg603!

bobbyg603 commented:
Ready to merge!
